crates/sparrow-instructions/src/evaluators/map/get_large_string_to_primitive.rs

```diff
@@ -0,0 +1,124 @@
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use arrow::array::{Array, ArrayRef, LargeStringArray, PrimitiveArray, StringArray};
+use arrow::datatypes::{ArrowPrimitiveType, DataType};
+use itertools::Itertools;
+use owning_ref::ArcRef;
+use sparrow_arrow::downcast::{downcast_primitive_array, downcast_string_array};
+
+use sparrow_plan::ValueRef;
+
+use crate::{Evaluator, EvaluatorFactory, StaticInfo};
+
+/// Evaluator for `get` on maps for large string (LargeUtf8) keys and primitive values.
+#[derive(Debug)]
+pub(in crate::evaluators) struct GetLargeStringToPrimitiveEvaluator<T>
+where
+    T: ArrowPrimitiveType + Sync + Send,
+{
+    map: ValueRef,
+    key: ValueRef,
+    // Make the compiler happy by using the type parameter
+    _phantom: PhantomData<T>,
+}
+
+impl<T> EvaluatorFactory for GetLargeStringToPrimitiveEvaluator<T>
```

Comment:
The only difference between these is the size of the length (`i32` vs. `i64`). Make that another type parameter and then we won't need the duplicate implementations?

Reply:
Yes, parameterized and remove this